modules/src/core/ics04_channel/packet.rs

```diff
@@ -110,6 +110,46 @@ pub struct Packet {
     pub timeout_timestamp: Timestamp,
 }
 
+struct PacketData<'a>(&'a [u8]);
+
+impl<'a> ::core::fmt::Debug for PacketData<'a> {
+    fn fmt(&self, formatter: &mut ::core::fmt::Formatter<'_>) -> Result<(), ::core::fmt::Error> {
+        write!(formatter, "{:?}", self.0)
+    }
+}
+
+impl ::core::fmt::Debug for Packet {
+    fn fmt(&self, formatter: &mut ::core::fmt::Formatter<'_>) -> Result<(), ::core::fmt::Error> {
+        // Remember: if you alter the definition of `Packet`,
+        // 1. update the formatter debug struct builder calls (return object of
+        //    this function)
+        // 2. update this destructuring assignment accordingly
+        let Packet {
+            sequence: _,
+            source_port: _,
+            source_channel: _,
+            destination_port: _,
+            destination_channel: _,
+            data,
+            timeout_height: _,
+            timeout_timestamp: _,
+        } = self;
```

Comment:
Good, I didn't think of using this guard statement!

Reply:
If all you want is the `data` field, this can be written more concisely as `let Packet { data, .. } = self;`

Reply:
Here, the "verbosity" is intentional: See [@hu55a1n1 's suggestion](https://github.com/informalsystems/ibc-rs/pull/1944#discussion_r823489356).

Basically we want the compiler to yell if we change any field in `Packet`, so that we can remember to update the debug output.

Reply:
That makes sense. And a pretty clever way to achieve that 😄